app/controllers/incomeSources/manage/ManageIncomeSourceDetailsController.scala

```diff
@@ -115,27 +117,53 @@ class ManageIncomeSourceDetailsController @Inject()(val view: ManageIncomeSource
   def showSoleTraderBusiness(id: String): Action[AnyContent] = (checkSessionTimeout andThen authenticate andThen retrieveNino
     andThen retrieveIncomeSources andThen retrieveBtaNavBar).async {
     implicit user =>
-      handleRequest(
-        sources = user.incomeSources,
-        isAgent = false,
-        backUrl = controllers.incomeSources.manage.routes.ManageIncomeSourceController.show(false).url,
-        id = Some(id),
-        incomeSourceType = SelfEmployment
-      )
+      withIncomeSourcesFS {
+        val result = handleRequest(
+          sources = user.incomeSources,
+          isAgent = false,
+          backUrl = controllers.incomeSources.manage.routes.ManageIncomeSourceController.show(false).url,
+          id = Some(id),
+          incomeSourceType = SelfEmployment
+        )
+
+        result.flatMap(
```

Comment:
I would think that we need to attempt save value first and then try evaluate page we have so far, 
so the order of  operations within withIncomeSourcesFS need to be changes;
* also, this might be not in the scope of this story, but I might assume that to evaluate page we need to read incomeSourceId from Mongo? not sure about this

Reply:
again, not critical but order should be: 1. save to Mongo 2. render page if success